gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/ProjectInfo.java

```diff
@@ -0,0 +1,310 @@
+    private ResourceId parent;
+
+    Builder() {
+      labels = new HashMap<>();
+    }
+
+    /**
+     * Set the user-assigned name of the project.
+     *
+     * This field is optional and can remain unset. Allowed characters are: lowercase and uppercase
+     * letters, numbers, hyphen, single-quote, double-quote, space, and exclamation point. This
+     * field can be changed after project creation.
+     */
+    public Builder name(String name) {
+      this.name = name;
+      return this;
+    }
+
+    /**
+     * Set the unique, user-assigned ID of the project.
+     *
+     * The ID must be 6 to 30 lowercase letters, digits, or hyphens. It must start with a letter.
+     * Trailing hyphens are prohibited. This field cannot be changed after the server creates the
+     * project.
+     */
+    public Builder id(String id) {
+      this.id = id;
+      return this;
+    }
+
+    /**
+     * Add a label associated with this project.
+     *
+     * See {@link #labels} for label restrictions.
+     */
+    public Builder addLabel(String key, String value) {
+      this.labels.put(key, value);
+      return this;
+    }
+
+    /**
+     * Remove a label associated with this project.
+     */
+    public Builder removeLabel(String key) {
+      this.labels.remove(key);
+      return this;
+    }
+
+    /**
+     * Clear the labels associated with this project.
+     */
+    public Builder clearLabels() {
+      this.labels.clear();
+      return this;
+    }
+
+    /**
+     * Set the labels associated with this project.
+     *
+     * Label keys must be between 1 and 63 characters long and must conform to the following regular
+     * expression: [a-z]([-a-z0-9]*[a-z0-9])?. Label values must be between 0 and 63 characters long
+     * and must conform to the regular expression ([a-z]([-a-z0-9]*[a-z0-9])?)?. No more than 256
+     * labels can be associated with a given resource. This field can be changed after project
+     * creation.
+     */
+    public Builder labels(Map<String, String> labels) {
+      this.labels = Maps.newHashMap(checkNotNull(labels));
+      return this;
+    }
+
+    Builder number(Long number) {
+      this.number = number;
+      return this;
+    }
+
+    Builder state(State state) {
+      this.state = state;
+      return this;
+    }
+
+    Builder createTimeMillis(Long createTimeMillis) {
+      this.createTimeMillis = createTimeMillis;
+      return this;
+    }
+
+    /**
+     * Set the parent of the project.
+     *
+     * If this field is left unset in a project creation request, the server will set this field by
+     * default to the creator of the project. The parent cannot be changed after the server creates
+     * the project. When calling {@link ResourceManager#replace}, be sure to set the parent of the
+     * new ProjectInfo instance. Leaving the parent unset or setting it to null in a replace request
+     * will cause an error.
+     */
+    public Builder parent(ResourceId parent) {
```

Comment:
So do we already support parents (and if so, what types?). If we don't support it yet, maybe
it should be a read-only property until we do.